server/etcdmain/grpc_proxy.go

```diff
@@ -461,32 +460,32 @@ func newGRPCProxyServer(lg *zap.Logger, client *clientv3.Client) *grpc.Server {
 	electionp := grpcproxy.NewElectionProxy(client)
 	lockp := grpcproxy.NewLockProxy(client)
 
-	alwaysLoggingDeciderServer := func(ctx context.Context, fullMethodName string, servingObject any) bool { return true }
+	srvMetrics := grpc_prometheus.NewServerMetrics()
+	prometheus.Register(srvMetrics)
 
 	grpcChainStreamList := []grpc.StreamServerInterceptor{
-		grpc_prometheus.StreamServerInterceptor,
+		srvMetrics.StreamServerInterceptor(),
 	}
 	grpcChainUnaryList := []grpc.UnaryServerInterceptor{
-		grpc_prometheus.UnaryServerInterceptor,
+		srvMetrics.UnaryServerInterceptor(),
 	}
+
 	if grpcProxyEnableLogging {
+		opts := []grpc_logging.Option{
+			grpc_logging.WithLogOnEvents(grpc_logging.StartCall, grpc_logging.FinishCall),
+		}
+
 		grpcChainStreamList = append(grpcChainStreamList,
-			grpc_ctxtags.StreamServerInterceptor(),
-			grpc_zap.PayloadStreamServerInterceptor(lg, alwaysLoggingDeciderServer),
+			grpc_logging.StreamServerInterceptor(interceptorLogger(lg), opts...),
```

Comment:
Looks like the old `grpc_ctxtags.StreamServerInterceptor` added a field for `peer.address`, https://github.com/grpc-ecosystem/go-grpc-middleware/blob/d42ae9d517069c2bd7f9339147a0eafa86b3d4a3/tags/interceptors.go#L70-L76.

The new logging interceptor adds this field as well, hence why I've removed the old grpc_ctxtags interceptors: 
https://github.com/grpc-ecosystem/go-grpc-middleware/blob/7da22cf3f3d3ae190467d9c7a3ea749b3d0e63b5/interceptors/logging/interceptors.go#L168-L170